controllers/nodeagent.go

```diff
@@ -61,17 +69,32 @@ func getNodeAgentObjectMeta(r *DPAReconciler) metav1.ObjectMeta {
 	}
 }
 
-func (r *DPAReconciler) ReconcileResticDaemonset(log logr.Logger) (bool, error) {
+func (r *DPAReconciler) ReconcileNodeAgentDaemonset(log logr.Logger) (bool, error) {
 	dpa := oadpv1alpha1.DataProtectionApplication{}
+	var deleteDaemonSet bool = false
+
 	if err := r.Get(r.Context, r.NamespacedName, &dpa); err != nil {
 		return false, err
 	}
-
 	// Define "static" portion of daemonset
 	ds := &appsv1.DaemonSet{
 		ObjectMeta: getNodeAgentObjectMeta(r),
 	}
+
+	if dpa.Spec.Configuration.Restic != nil {
+		// V(-1) corresponds to the warn level
+		var deprecationMsg string = "(Deprecation Warning) Use nodeAgentConfig instead of restic, which is deprecated and will be removed with the OADP 1.4"
+		log.V(-1).Info(deprecationMsg)
+		r.EventRecorder.Event(&dpa, corev1.EventTypeWarning, "DeprecationResticConfig", deprecationMsg)
+	}
+
 	if dpa.Spec.Configuration.Restic == nil || dpa.Spec.Configuration.Restic != nil && (dpa.Spec.Configuration.Restic.Enable == nil || !*dpa.Spec.Configuration.Restic.Enable) {
```

Comment:
This logic is br8ken, fixing...

Reply:
it is working for restic in 1.2

Reply:
That was comment to myself, but the issue with not creating DaemonSet was around my logic in if/else for the nil pointer check. I had:

```
if dpa.Spec.Configuration.Restic != nil && dpa.Spec.Configuration.Restic.blahblah != nil .... {
} else if dpa.Spec.Configuration.NodeAgent.blahblah != nil {
}
```

So that was wrong, because if `Restic` was used and we were missing just `Restic.blahblah`, then the check on the similar attribute for `NodeAgent.blahblah` was done, however this didn't mean the `NodeAgent` was used at all.

Fixed and now the `DaemonSet` is being created properly. Will upload new rev. shortly.

Reply:
It's fixed now.